ament_tools/build_types/cmake.py

```diff
@@ -216,6 +217,39 @@ def _common_cmake_on_test(self, context, build_type):
                 context.ctest_args)
 
     def on_install(self, context):
+        # First determine the files being deployed with skip_if_exists=True and remove them.
+        environment_hooks_path = \
+            os.path.join('share', context.package_manifest.name, 'environment')
+
+        environment_hooks_to_be_deployed = []
+
+        # Prepare to deploy PATH environment hook
+        ext = '.sh' if not IS_WINDOWS else '.bat'
+        path_template_path = get_environment_hook_template_path('path' + ext)
+        environment_hooks_to_be_deployed.append(path_template_path)
+        environment_hooks = [os.path.join(environment_hooks_path, 'path' + ext)]
+
+        # Prepare to deploy library path environment hook if not on Windows
+        if not IS_WINDOWS:
+            library_template_path = get_environment_hook_template_path('library_path.sh')
+            environment_hooks_to_be_deployed.append(library_template_path)
+            environment_hooks.append(os.path.join(environment_hooks_path, 'library_path.sh'))
+
+        # Expand package level setup files
+        destinations = \
+            expand_package_level_setup_files(context, environment_hooks, environment_hooks_path)
+
+        # Remove package level setup files so they can be replaced correctly either in the
+        # cmake install step or later with deploy_file(..., skip_if_exists=True)
+        for destination in destinations:
+            destination_path = compute_deploy_destination(
+                context,
+                os.path.basename(destination),
+                os.path.dirname(os.path.relpath(destination, context.build_space))
+            )
+            if os.path.exists(destination_path):
```

Comment:
This is not enough if the `destination_path` is a link pointing to a non existing file.